modules/progress/settings.go

```diff
@@ -0,0 +1,61 @@
+package progress
+
+import (
+	"os"
+
+	"github.com/olebedev/config"
+	"github.com/wtfutil/wtf/cfg"
+)
+
+const (
+	defaultFocusable = false
+	defaultTitle     = "Progress"
+)
+
+type colors struct {
+	gradientA string `help:"Start color for linear gradient." values:"any X11 or hex color" optional:"true" default:"#56ab2f"`
+	gradientB string `help:"End color for linear gradient." values:"any X11 or hex color" optional:"true" default:"#a8e063"`
+	solid     string `help:"Use a solid color instead of linear color gradient ." values:"any X11 or hex color" optional:"true"`
+}
+
+// Settings defines the configuration properties for this module
+type Settings struct {
+	colors
+	common *cfg.Common
+
+	showPercentage bool `help:"Wether or not to display percentage next to the progress bar." values:"true or false" optional:"true" default:"true"`
+	padding        int  `help:"Amount of spaces to add as left/right padding." values:"A positive integer, 0..n" optional:"true" default:"1"`
+
+	minimum float64 `help:"Minimum progress value." values:"A positive decimal value, 0.0..n.n" optional:"true" default:"0"`
+	maximum float64 `help:"Maximum progress value." values:"A positive decimal value, 0.0..n.n" optional:"true" default:"0"`
+	current float64 `help:"Current progress value. If maximum value is 0, current value is assumed to be a percentage between 0-100." values:"A positive decimal value, 0.0..n.n" optional:"true" default:"0"`
+
+	minimumCmd string `help:"Execute shell command to determine minimum progress value. Return value must be numeric." values:"Any shell command" optional:"true"`
+	maximumCmd string `help:"Execute shell command to determine maximum progress value. Return value must be numeric." values:"Any shell command" optional:"true"`
+	currentCmd string `help:"Execute shell command to determine current progress value. Return value must be numeric." values:"Any shell command" optional:"true"`
+
+	shell string `help:"Shell to use for executing value commands." values:"shell path" optional:"true" default:"$SHELL environment variable"`
```

Comment:
Curious about this option. Are we expecting different shells to produce the same results for the same commands? Since you can call arbitrary commands, wondering if it makes more sense to wrap results in shell scripts as a driver for alternate shell usage.

Reply:
I don't think there would be different results, unless someone uses functions that are specific to, e.g., zsh, but then they probably use that as their default `$SHELL` anyway. I added it as a setting just because it was easy. 😄 It sounds like a good idea to instruct users to define shell scripts if they want an alternative shell, though. 👍